sips/network_topology.md

```diff
@@ -0,0 +1,120 @@
+|     Author     |           Title            |  Category  |       Status        |    Date    |
+| -------------- | -------------------------- | ---------- | ------------------- | ---------- |
+| Matheus Franco | Network Topology           | Network    | open-for-discussion | 2024-03-06 |
+
+## Table of Contents <!-- omit from toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Spec Change](#spec-change)
+- [Alternative Solutions](#alternative-solutions)
+	- [1. Greedy Algorithm](#1-greedy-algorithm)
+	- [2. MaxReach](#2-maxreach)
+	- [3. LowestID](#3-lowestid)
+- [Comparisons](#comparisons)
+
+
+## Summary
+
+This SIP proposes to useMinHash for assigning committees to topics, aiming to reduce the number of irrelevant (non-committee) messages operators must process.
+
+## Motivation
+
+Currently, committee-to-topic assignment is random, using a hash of committee operators modulo 128 (number of topics):
+
+```go
+// Bytes -> Int -> Modulo 128
+func CommitteeSubnet(cid spectypes.CommitteeID) uint64 {
+	subnet := new(big.Int).Mod(new(big.Int).SetBytes(cid[:]), bigIntSubnetsCount)
+	return subnet.Uint64()
+}
+
+// Operators -> Bytes -> Hash
+func GetCommitteeID(committee []OperatorID) CommitteeID {
+	// sort
+	sort.Slice(committee, func(i, j int) bool {
+		return committee[i] < committee[j]
+	})
+	// Convert to bytes
+	bytes := make([]byte, len(committee)*4)
+	for i, v := range committee {
+		binary.LittleEndian.PutUint32(bytes[i*4:], uint32(v))
+	}
+	// Hash
+	return sha256.Sum256(bytes)
+}
+```
+
+This randomness forces operators active in many committees to listen to multiple topics and process many irrelevant messages, which limits scalability. 
+
+In contrast, MinHash assigns committees to topics based on the minimum hash of the committee operators. This increases the probability of committees being assigned to the same topic when they have most of their operators in common.
+
+With MinHash, the probability of two committees getting the same topic is the number of shared operators over the number of total operators.
+
+## Spec Change
+For each committee, we assign a topic based on the minimum hash of the committee operators.
+
+```go
+func AssignCommitteeToTopic(committee []OperatorID) uint64 {
+	minHash := uint64(math.MaxUint64)
+	for _, id := range committee {
+		hash := sha256(id)
+		if hash < minHash {
+			minHash = hash
+		}
+	}
+	return minHash % NUM_TOPICS
+}
+```
+
+## Alternative Solutions
+
```

Comment:
since you decided to go to those details, maybe add the following:
1. Definition of the properties (so it is clear to the reader)
2. nice to have an introduction sentence: "these are the alternatives we explored"